Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -279,37 +275,37 @@ public Task<AndConstraint<TAssertions>> NotThrowAfterAsync(TimeSpan waitTime, Ti
             .BecauseOf(because, becauseArgs)
             .FailWith("Expected {context} not to throw any exceptions after {0}{reason}, but found <null>.", waitTime);
 
-        if (!success)
-        {
-            return Task.FromResult(new AndConstraint<TAssertions>((TAssertions)this));
-        }
-
-        return AssertionTaskAsync();
-
-        async Task<AndConstraint<TAssertions>> AssertionTaskAsync()
+        if (success)
         {
-            TimeSpan? invocationEndTime = null;
-            Exception exception = null;
-            ITimer timer = Clock.StartTimer();
+            return AssertionTaskAsync();
 
-            while (invocationEndTime is null || invocationEndTime < waitTime)
+            async Task<AndConstraint<TAssertions>> AssertionTaskAsync()
```

Comment:
You might still want to keep local functions at the bottom of the parent method. Effectively, they work as "private methods in a method scope", meaning treating them as if they were private puts the focus on the public code of the method first, which can improve readability: people tend to want to look at the high level concepts first, and then drill down into the lower level ones as needed.

The justification would be similar as to why the recommendation is to keep `private` methods below `public` ones in classes.